### ISSUE TITLE
Unable to compile version 3.0.0

### ISSUE THREAD

**Issue by marcschumacher** [zalando-stups/java-sproc-wrapper#71]
I am unable to compile the latest release (tag 3.0.0) of the library because of warnings. When removing `-werror` from the compiler options in `pom.xml` it works though.

```
~/src/zalando/java-sproc-wrapper on  HEAD (d10ad1b) on ☁️ schumi@zalando.de
❯ mvn clean compile
Picked up JAVA_TOOL_OPTIONS: -Djava.awt.headless=true
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------< org.zalando:zalando-sprocwrapper >------------------
[INFO] Building Stored Procedure Wrapper 3.0.0
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ zalando-sprocwrapper ---
[INFO] Deleting /Users/mschumacher/src/zalando/java-sproc-wrapper/target
[INFO]
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-maven) @ zalando-sprocwrapper ---
[INFO]
[INFO] --- duplicate-finder-maven-plugin:1.3.0:check (default) @ zalando-sprocwrapper ---
[INFO] Checking compile classpath
[INFO] Checking runtime classpath
[INFO] Checking test classpath
[INFO]
[INFO] --- jacoco-maven-plugin:0.8.4:prepare-agent (prepare-agent) @ zalando-sprocwrapper ---
[INFO] argLine set to -javaagent:/Users/mschumacher/.m2/repository/org/jacoco/org.jacoco.agent/0.8.4/org.jacoco.agent-0.8.4-runtime.jar=destfile=/Users/mschumacher/src/zalando/java-sproc-wrapper/target/jacoco.exec
[INFO]
[INFO] --- maven-resources-plugin:3.1.0:resources (default-resources) @ zalando-sprocwrapper ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ zalando-sprocwrapper ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 109 source files to /Users/mschumacher/src/zalando/java-sproc-wrapper/target/classes
[INFO] -------------------------------------------------------------
[WARNING] COMPILATION WARNING :
[INFO] -------------------------------------------------------------
[WARNING] /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/java/org/zalando/typemapper/postgres/PgTypeHelper.java:[268,23] isAccessible() in java.lang.reflect.AccessibleObject has been deprecated
[WARNING] /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/java/org/zalando/typemapper/postgres/PgTypeHelper.java:[420,61] newInstance() in java.lang.Class has been deprecated
[WARNING] /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/java/org/zalando/typemapper/postgres/PgTypeHelper.java:[435,81] newInstance() in java.lang.Class has been deprecated
[WARNING] /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/java/org/zalando/sprocwrapper/globalvaluetransformer/GlobalValueTransformerLoader.java:[74,81] newInstance() in java.lang.Class has been deprecated
[WARNING] /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/java/org/zalando/typemapper/core/fieldMapper/ValueTransformerFieldMapper.java:[11,49] newInstance() in java.lang.Class has been deprecated
[WARNING] /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/java/org/zalando/typemapper/core/TypeMapper.java:[62,37] newInstance() in java.lang.Class has been deprecated
[WARNING] /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/java/org/zalando/typemapper/core/Mapping.java:[236,50] newInstance() in java.lang.Class has been deprecated
[WARNING] /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/java/org/zalando/sprocwrapper/proxy/SProcCallHandler.java:[44,57] newInstance() in java.lang.Class has been deprecated
[WARNING] /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/java/org/zalando/sprocwrapper/proxy/SProcCallHandler.java:[116,82] newInstance() in java.lang.Class has been deprecated
[WARNING] /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/java/org/zalando/sprocwrapper/proxy/SProcServiceAnnotationHandler.java:[61,86] newInstance() in java.lang.Class has been deprecated
[WARNING] /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/java/org/zalando/sprocwrapper/globalobjecttransformer/GlobalObjectTransformerLoader.java:[66,94] newInstance() in java.lang.Class has been deprecated
[WARNING] /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/java/org/zalando/sprocwrapper/dsprovider/BitmapShardDataSourceProvider.java:[91,50] newInstance() in java.lang.Class has been deprecated
[WARNING] /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/java/org/zalando/typemapper/core/fieldMapper/ObjectFieldMapper.java:[46,57] newInstance() in java.lang.Class has been deprecated
[WARNING] /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/java/org/zalando/typemapper/core/fieldMapper/ObjectFieldMapper.java:[70,27] newInstance() in java.lang.Class has been deprecated
[WARNING] /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/java/org/zalando/typemapper/core/fieldMapper/PrimitiveLongFieldMapper.java:[13,20] Long(long) in java.lang.Long has been deprecated
[WARNING] /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/java/org/zalando/typemapper/core/fieldMapper/PrimitiveLongFieldMapper.java:[22,16] Long(long) in java.lang.Long has been deprecated
[INFO] 16 warnings
[INFO] -------------------------------------------------------------
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/java/org/zalando/typemapper/postgres/PgTypeHelper.java: warnings found and -Werror specified
[INFO] 1 error
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  5.713 s
[INFO] Finished at: 2021-03-18T10:26:43+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project zalando-sprocwrapper: Compilation failure
[ERROR] /Users/mschumacher/src/zalando/java-sproc-wrapper/src/main/java/org/zalando/typemapper/postgres/PgTypeHelper.java: warnings found and -Werror specified
[ERROR]
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

**Comment by vadeg** [comment]
Hi @marcschumacher!

Which java version are you using? I have the same problem with Java 11, but with Java 8 it compiles correct.

**Comment by vadeg** [comment]
Fixed in #76
